### PR TITLE
Feature: Interest groups bubble plotting

### PR DESCRIPTION
**Pull request by amovar18**
## Description
In this PR, we aim to create a container for interest group bubbles. So that it looks better when there are more number of interest group bubbles.

## Relevant Technical Choices
- A new module `bubbles` has been added. This module will manage the animation for the loading the interest group and joining interest group.
- this will also manage the expansion and minification of the interest group bubble container.

## Testing Instructions
- Clone the branch.
- In the terminal run `cd ee && npm i && npm run start`.
- Now open https://localhost:8080/index.html.
- Once interest group is added by advertiser you can click on this and expand the circle to see interest groups.

## Screenshot/Screencast

Screenshot 2024-11-04 at 16 34 55

https://github.com/user-attachments/assets/2fe99e76-2c28-4cb6-8bd2-84c7507bccb2

---

## Checklist

- [x] I have thoroughly tested this code to the best of my abilities.
- [x] I have reviewed the code myself before requesting a review.
- [x] This code is covered by unit tests to verify that it works as intended.
- [x] The QA of this PR is done by a member of the QA team (to be checked by QA).

Partially addresses #857

### CODE REVIEW

**Comment by mohdsayed** [comment]
@amovar18 I think closing the bubble on play button is not intuitive enough. We should probably add any other icon indicating how the bubble will expand and close.

image

**Comment by mohdsayed** [comment]
3. The number is not very visible when there are multiple color bubbles.
image

**Comment by mohdsayed** [comment]
4. When the circle doesn't not have any bubble yet, and circle expands, it first goes to the left and then in the center

**Comment by mohdsayed** [comment]
5. Some bubbles are so small that they are not readable ( laptop in this case )
image

**Comment by mohdsayed** [comment]
6. Since the bubble are clickable, the cursor should change when hovering over it.

File: ee/src/config.js [hunk 1]
```diff
@@ -35,39 +36,53 @@ const config = {
...
+        interestGroups: ['shoes', 'heels', 'phones', 'laptops', 'blankets'],
       },
       {
         type: 'advertiser',
         website: 'adv2.com',
         datetime: '2023-10-01 11:00',
-        igGroupsCount: 2,
+        igGroupsCount: 7,
+        interestGroups: [
+          'wJk91XxpVnFvc2oySFZtdjFaQmxzV3U1ODdnPT18',
```
Comment (mohdsayed) [inline ee/src/config.js:48]: Let's use some name instead of ID.

**Review by mohdsayed — CHANGES_REQUESTED** [review]
Looks good, just few small improvements are required.

**Comment by mohdsayed** [comment]
7. I was thinking maybe we could add a subtle overlay in the background. When the circle expands, it would appear more focused.

File: ee/src/config.js (hunk 1 above)
Comment (amovar18) [reply ee/src/config.js]: Fixed in latest commits.

**Comment by amovar18** [comment]
@mohdsayed fixed the UX issues in latest commits.

**Comment by mohdsayed** [comment]
@amovar18 one of the build is failing.

**Comment by amovar18** [comment]
Fixed the linting test.

**Comment by mohdsayed** [comment]
Some more improvements are required.

- [ ]  Remove blur from small circle.
- [ ]  Make expand and collapse icons grey.
- [ ]  Move collapse icon closer to the larger circle.
- [ ]  Close on ESC key press and when clicking outside the larger circle.
- [ ]  Use grey color for both small and large circles.
- [ ]  Change cursor to pointer for all clickable circles.
- [ ]  Fix issue where expand and collapse icons appear after a quick refresh.

**Comment by mohdsayed** [comment]
@amovar18 Two issues are still pending

https://github.com/GoogleChromeLabs/ps-analysis-tool/pull/865#issuecomment-2477881487

File: packages/explorable-explanations/public/index.html [hunk 2]
```diff
@@ -47,12 +53,108 @@
...
+
+        #count-display {
+            font-size: 16px;
+            position: absolute;
+            margin: 0px;
+            backdrop-filter: blur(4px);
+            font-weight: bolder;
+            border-radius: 50%;
+        }
+
```
Comment (mohdsayed) [inline packages/explorable-explanations/public/index.html:157]: As the CSS is growing, we should soon move it to a css file.

File: packages/explorable-explanations/src/lib/bubbles.js [hunk 3]
```diff
@@ -0,0 +1,525 @@
+/*
```
Comment (mohdsayed) [inline packages/explorable-explanations/src/lib/bubbles.js:1]: Moving this to module as it encapsulates functional unit of the application, responsible for a specific feature.

**Review by mohdsayed — APPROVED** [review]
Looks great

**Pull request merged**